provisioning/miniprov/hare_mp/main.py

```diff
@@ -343,6 +335,13 @@ def main():
         filename = parsed.filename[0] or '/var/lib/hare/cluster.yaml'
         save(filename, generate_cdf(url))
 
+    if parsed.init:
```

Comment:
I think we can avoid these multiple if loops by setting `nargs` to a non-zero value, can you try https://docs.python.org/3/library/argparse.html?
This automatically adds a check too on the args.

Reply:
I would suggest to address #1482 together with this PR.

`init` or `config` would be a command while `--filename` is an option to a command. Otherwise we have cross-dependencies between options (`--init` handler is required to look for `--filename`) which doesn't look good and leads to over-complication in the code

Reply:
Yep, I agree with @knekrasov.